aten/src/ATen/native/quantized/cpu/qnnpack/src/deconv-run.cc

```diff
@@ -19,7 +83,242 @@ enum pytorch_qnnp_status qnnpackDeConv(
     const uint8_t output_zero_point,
     const bool transpose,
     uint8_t* output,
-    pthreadpool_t threadpool);
+    pthreadpool_t threadpool) {
+
+  if (batch_size == 0) {
+    // Doesn't matter what's going on, if no batches, return
+    return pytorch_qnnp_status_success;
+  }
+
+  // Check all invalid parameters
+  const size_t kernel_width = deconv_p.kernel_dims[0];
+  const size_t kernel_height = deconv_p.kernel_dims[1];
+  if (kernel_width == 0 || kernel_height == 0) {
+    pytorch_qnnp_log_error(
+        "failed to create deconvolution with %" PRIu32 "x%" PRIu32
+        " kernel: kernel dimensions must be non-zero",
+        (uint32_t)kernel_width,
+        (uint32_t)kernel_height);
+    return pytorch_qnnp_status_invalid_parameter;
+  }
+
+  const size_t stride_width = deconv_p.stride_dims[0];
+  const size_t stride_height = deconv_p.stride_dims[1];
+  if (stride_width == 0 || stride_height == 0) {
+    pytorch_qnnp_log_error(
+        "failed to create deconvolution with %" PRIu32 "x%" PRIu32
+        " stride: "
+        "stride dimensions must be non-zero",
+        (uint32_t)stride_width,
+        (uint32_t)stride_height);
+    return pytorch_qnnp_status_invalid_parameter;
+  }
+
+  const size_t dilation_width = deconv_p.dilation_dims[0];
+  const size_t dilation_height = deconv_p.dilation_dims[1];
+  if (dilation_width == 0 || dilation_height == 0) {
+    pytorch_qnnp_log_error(
+        "failed to create deconvolution with %" PRIu32 "x%" PRIu32
+        " dilation: "
+        "dilation dimensions must be non-zero",
+        (uint32_t)dilation_width,
+        (uint32_t)dilation_height);
+    return pytorch_qnnp_status_invalid_parameter;
+  }
+
+  if (input_scale <= 0.0f || !isnormal(input_scale)) {
+    pytorch_qnnp_log_error(
+        "failed to create deconvolution with %.7g input scale: "
+        "scale must be finite and positive",
+        input_scale);
+    return pytorch_qnnp_status_invalid_parameter;
+  }
+
+  const float kernel_scale = deconv_p.kernel_scale;
+  if (kernel_scale <= 0.0f || !isnormal(kernel_scale)) {
+    pytorch_qnnp_log_error(
+        "failed to create deconvolution with %.7g kernel scale: "
+        "scale must be finite and positive",
+        kernel_scale);
+    return pytorch_qnnp_status_invalid_parameter;
+  }
+
+  if (output_scale <= 0.0f || !isnormal(output_scale)) {
+    pytorch_qnnp_log_error(
+        "failed to create deconvolution with %.7g output scale: "
+        "scale must be finite and positive",
+        output_scale);
+    return pytorch_qnnp_status_invalid_parameter;
+  }
+
+  // Check all unsupported parameters
+  const float deconvolution_scale = input_scale * kernel_scale / output_scale;
+  if (deconvolution_scale >= 1.0f) {
```

Comment:
@kimishpatel does the float requantization PR apply to the deconv operation too?

Reply:
Yes it does. And it is covered under conv operator. Please see this PR: https://github.com/pytorch/pytorch/pull/35856. I suggest we wait for it to land? Else I will have to make fixes here.

Reply:
Adding a TODO in here in case this lands before #35856

Reply:
#35856 has landed.

Reply:
Can you elaborate what changes the PR will require?

Reply:
I was gonna say like this: https://github.com/pytorch/pytorch/blob/master/aten/src/ATen/native/quantized/cpu/qnnpack/src/conv-run.cc#L308. But it seems that I missed removing this check since it is not longer a requirement. @supriyar, did this not trigger on the model we ran the PR for requant changes? Seems strange.
@z-a-f I think you can remove this check.

Reply:
>> @supriyar, did this not trigger on the model we ran the PR for requant changes? Seems strange.

I had made the changes locally when I ran the model so wasn't an issue.

Reply:
@kimishpatel I am confused, we don't need this condition anymore? Can I just remove the if statement defined in line 156 completely?

Reply:
Yes, we dont need this condition anymore. Please remove it. I need to remove similar ones from conv-run and fc-run.

Reply:
Will do before landing